proto/catalog.proto

```diff
@@ -176,7 +178,8 @@ message Table {
   uint32 database_id = 3;
   string name = 4;
   repeated plan_common.ColumnCatalog columns = 5;
-  repeated common.ColumnOrder pk = 6;
+  plan_common.DefaultColumns default_columns = 6;
```

Comment:
We're not allowed to break the backward compatibility here. For example, we'll fail to deserialize the catalog persisted by an old version of RisingWave with the new version. Please use a new field number here.

Reply:
BTW, do you think it's better to put the info of default column to `ColumnDesc` and be consistent with `GeneratedColumnDesc`?

Reply:
That was my first implementation, but later on I found many unnecessary works were to be done with that method. Do you prefer consistency here over ... like ... cleanness and redundancy here?

Reply:
IMO default_columns are always presented in a somehow `Vec<IndexAndExpr>` way

Reply:
I'd prefer putting it together with "generated column". 🥵 Actually they should even reside in an `oneof`.